crates/ruff_python_parser/src/parser/tests/function.rs

```diff
@@ -0,0 +1,79 @@
+#[cfg(test)]
+mod tests {
+    use crate::lexer::LexicalErrorType;
+    use crate::parser::parse_suite;
+    use crate::ParseErrorType;
+
+    macro_rules! function_and_lambda {
+        ($($name:ident: $code:expr,)*) => {
+            $(
+                #[test]
+                fn $name() {
+                    let parse_ast = crate::parser::parse_suite($code);
+                    insta::assert_debug_snapshot!(parse_ast);
+                }
+            )*
+        }
+    }
```

Comment:
I try to avoid macros for test generation (okay, not true, I try to avoid macros in general, it's a last resort option in my view) because they make tests harder to read and they don't work well with IDEs and other dev tools (e.g. jumping to the failing test doesn't work). 

Not using macros will require some more boilerblate but that's a worth trade in my view, considering that it makes the tests more familiar to readers

```rust

fn parse_program(code: &str) -> Result<Suite, ParseError> {
    crate::parser::parse_suite(code)
}

#[test]
fn test_function_no_args_with_ranges() {
    let parsed = parse_program("def f(): pass")
    assert_debug_snapshot!(parsed)
}
```

The `parse_program` helper function is probably overkill. But a utility function can be helpful in case we need to perform more setup logic (to avoid code duplication)

Reply:
Good point, those macros in the tests were already there, I will remove them.